bitcoin/src/sign_message.rs

```diff
@@ -108,11 +108,10 @@ mod message_signing {
             serialized
         }
 
-        /// Create from a byte slice.
-        pub fn from_slice(bytes: &[u8]) -> Result<MessageSignature, MessageSignatureError> {
-            if bytes.len() != 65 {
-                return Err(MessageSignatureError::InvalidLength);
-            }
+        /// Creates a `MessageSignature` from a fixed-length array.
+        pub fn from_byte_array(
+            bytes: &[u8; 65],
+        ) -> Result<MessageSignature, MessageSignatureError> {
```

Comment:
Oh, I've just realized the `InvalidLength` variant is not possible here so the error type should be different.

Reply:
I have removed `InvalidLength` and instead used `secp256k1::Error::InvalidMessage` in the two cases where it was used.

Or would it be better to leave that in and change `from_byte_array` to return a `secp256k1::Error` and then in `from_slice` convert the error e.g. `Self::from_byte_array(&byte_array).map_err(MessageSignatureError::from)`?

Reply:
Yes, the latter would be better.

Reply:
I have changed it to the latter